### ISSUE TITLE
canu failed with 'failed to create the overlap store'.

### ISSUE THREAD

**Issue by lguzmanz** [marbl/canu#110]
Trying to run the following command:

$ canu -trim -p PacBio10_trimmed -d PacBio10_trimmed genomeSize=200000 -pacbio-corrected /work/03868/lguzmanz/liz_trials/PacBio_correctedreads/corrected_Canu_pacbio10.correctedReads.fasta.gz "useGrid=false"

Got the following error message: 

```
ERROR:
ERROR:  Failed with exit code 1.
ERROR:
================================================================================
Don't panic, but a mostly harmless error occurred and canu failed.

Disk space available:  17025761.9 GB

Last 50 lines of the relevant log file (/work/03868/lguzmanz/liz_trials/april_20/PacBio10_trimmed/trimming/PacBio10_trimmed.ovlStore.err):

Summing overlap counts for 1 reads from '/work/03868/lguzmanz/liz_trials/april_20/PacBio10_trimmed/trimming/1-overlapper/001/000001.counts'.
Found no overlaps to sort.

canu failed with 'failed to create the overlap store'.
```

I had to manually correct 18 PacBio reads because the correct/trim/assemble default canu code ran into issues, now that I look back on it, it was probably the same error with the "failed to create overlap store" when trimming. 

Any help would be greatly appreciated!

**Comment by brianwalenz** [comment]
This certainly seems like it found _one_ read in the input to the trimming stage.  Look for hints or errors in the correction/ logs.  A good place to start would be correction.html.

It is possible that all the corrected reads were short - the default minimum is 1000bp.

**Comment by brianwalenz** [comment]
When you say "manually correct 18 PacBio reads" - did you really mean _reads_ and not _jobs_?  Can you summarize this process?

PS - three backtics (usually above the tab key) to get blocks of text formatted as, well, blocks of text.

**Comment by lguzmanz** [comment]
Hi brianwalenz, 

What I meant by manually was that by default the "canu" command will correct, trim and assemble a set of reads but because it was giving me issues I thought about doing each step individually (http://canu.readthedocs.org/en/stable/quick-start.html). I basically issued the "canu -correct" command for each of the PacBio file of reads I have, which are 18 total. 

I do not see any errors or hints in the correction.html files, can you give me more specifics into what I should be looking out for? 

I don't think the length of the reads is the issue, within the pacbio10.correctedReads.fasta.gz file there are 4 reads of the following length: 5,381/ 1,122/ 1,201/ 2,542......

**Comment by brianwalenz** [comment]
Ah, you need to run all reads (all files) at the same time.  There isn't enough coverage in each file to do much of anything.  I'm surprised you got any corrected reads at all.

canu will happily take wildcards -- -pacbio-raw *fastq.

**Comment by skoren** [comment]
Based on your genome size, 200000 (FYI Canu understand suffixes so you can write 200k or 2m), and 18 corrected reads, even assuming the reads average 10Kb each, you have <1X coverage so finding no overlaps makes sense. It looks from the one corrected sample you posted above your correct read average length is 2500 giving you < 0.5X.

How much coverage do you have in raw data that you're correcting? If you have less than 20X you're not going to get much from the correction.

**Comment by georgiab7** [comment]
Hi,  

I have received a similar error when trying to run Canu, yet haven't been able to solve it after reading this thread. 
A snippet of my output with my error is below: 

ERROR:
ERROR:  Failed with exit code 139.  (rc=35584)
ERROR:
================================================================================
Don't panic, but a mostly harmless error occurred and canu failed.

Disk space available:  132.637 GB

Last 50 lines of the relevant log file (/Users/georgia/honoursproject/F10/F10_1500_canu/F10_1500/correction/F10_1500.ovlStore.err):

Found 44100942 (44.10 million) overlaps.
Configuring for 2.00 GB to 8.00 GB memory and 240 open files.
Will sort using 1 files; 58720256 (58.72 million) overlaps per bucket; 2.00 GB memory per bucket
  bucket   1 has 44100942 olaps.
  bucket   2 has 0 olaps.
Will sort 44.101 million overlaps per bucket, using 1 buckets 1.56 GB per bucket.


-- BUCKETIZING --

-  Bucketizing '/Users/georgia/honoursproject/F10/F10_1500_canu/F10_1500/correction/1-overlapper/results/000001.ovb'

canu failed with 'failed to create the overlap store'.
<img width="783" alt="screen shot 2017-05-12 at 2 52 28 pm" src="https://cloud.githubusercontent.com/assets/25476884/25980505/14bdb9cc-3723-11e7-87cf-eaa41bd2657a.png">


I have looked up the error code and it is a "segmentation fault" however I'm very new to this field and don't know how to progress from here. 

Any help or suggestions would be greatly appreciated and apologies if this is a simple oversight on my behalf.

**Comment by brianwalenz** [comment]
First, please create a new issue for this - just cut-n-paste what you have above into it.

I don't know why it's failing, but I do see there are only 984 overlaps, which is quite low for a 5mbp genome.  For comparison, ecoli with very long nanopore reads has 62,000 overlaps.  How many reads were in the input?  What is the coverage?

Paste in some more of the screen output (or attach as a file).

This also looks like an older version of canu (canu -version), so consider upgrading.

**Comment by georgiab7** [comment]
Hi Brian, 

Thanks for your help, I updated canu and was able to run it with the same data set I used above. 
I could still create a new issue but for me this problem was solved by the update so I don't see much value in doing that. 

Georgia

**Comment by brianwalenz** [comment]
Awesome!

**Comment by drs357** [comment]
I have the same error using raw nano-pore reads as it found no overlaps to sort, is it possible that there are no overlaps? Any suggestions to fix this?

Thanks
